apps/dgst.c

```diff
@@ -461,7 +461,7 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
     size_t len;
     int i, backslash = 0;
 
-    for (;;) {
+    for (i = 0; !BIO_eof(bp);) {
```

Comment:
from the style POV, I would prefer `while (!BIO_eof(bp)) {`
since i is dead after this loop.

Reply:
done